### PR TITLE
Adding translation for cn

### PR DESCRIPTION
**Pull request by tuantuancc**

### CODE REVIEW

File: v2/cn/getting-started.md [hunk 1]
```diff
@@ -12,12 +12,24 @@ First of all, please go to the [Download](download.html) page to get the source
 code tarball of [OpenResty](openresty.html), and see the [Installation](installation.html) page
```
Comment (agentzh) [inline v2/cn/getting-started.md]: Lacking translation for this paragraph? ;)

File: v2/cn/getting-started.md [hunk 2]
```diff
@@ -12,12 +12,24 @@ First of all, please go to the [Download](download.html) page to get the source
...
 for how to build and install it into your system.
 
+First of all, you need to install OpenResty to your system.
+首先，你需要将 OpenResty 安装到你的系统。
+
+If you are in Linux, please check out OpenResty's official pre-built packages if your Linux distribution is currently supported. If you use the official pre-built package, please replace the nginx command in this document below with the openresty command.
+如果你使用的是 Linux 系统，请先用 OpenResty 的官方预装包，检查你的 Linux 发行版是否支持。如果你使用官方预装包，请在下面的文件里，将 Nginx 命令改为 OpenResty 命令。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: 1. "pre-built packages" => "预编译包"
2. "if" is not translated in the first sentence?
3. with the openresty command. Here openresty is case sensitive. do not change it to `OpenResty`.

Comment (agentzh) [inline v2/cn/getting-started.md]: We need links here on the "pre-built packages". I think the original English markdown file has it?

For example, `[pre-built packages](linux-packages.html)` is a markdown link. please refer to the markdown syntax manual for more details on the markdown notation. thank you!

File: v2/cn/getting-started.md [hunk 3]
```diff
@@ -12,12 +12,24 @@ First of all, please go to the [Download](download.html) page to get the source
...
+
+Failing that, you can go to the Download page to get the source code tarball of OpenResty, and see the Installation page for how to build and install it into your system.
+如果你的 Linux 系统不支持，你可以到下载页面，获取 OpenResty 的源代码tar压缩包，然后查看安装页面里关于如何安装的信息。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: 1. "source code tarball" => "源码压缩包"
2. need a link on the "Download page".
3. Need a link on the "Installation page".

File: v2/cn/getting-started.md [hunk 4]
```diff
@@ -12,12 +12,24 @@ First of all, please go to the [Download](download.html) page to get the source
...
+
 
 # HelloWorld
 
 ## Prepare directory layout
+## 准备好目录
+
 We first create a separate directory for our experiments. You can use an arbitrary
 directory. Here for simplicity, we just use `~/work`:
+首先，我们要新建一个单独的目录。你可以使用任何目录，简化起见，在这里我们使用 ~/work:
```
Comment (agentzh) [inline v2/cn/getting-started.md]: We need the backticks around `~/work`.

File: v2/cn/getting-started.md [hunk 5]
```diff
@@ -27,10 +39,14 @@ mkdir logs/ conf/
 
 Note that we've also created the `logs/` directory for logging files and `conf/` for
 our config files.
+需要注意的是，我们还需要为日志文件建立 logs/ 目录，为配置文件建立 conf/ 目录。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: Do not omit backticks since they are for formatting and significant.

File: v2/cn/getting-started.md [hunk 6]
```diff
@@ -55,46 +71,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 of all the exisitng goodies in the [Nginx](nginx.html) world.
+如果你很了解 Nginx 的布局，上面的内容你会觉得很熟悉。OpenResty 只是一个借助于插件模块来实现的 Nginx 加强版。你可以利用 Nginx 世界里所有的好东西。
 
 ## Start the [Nginx](nginx.html) server
+## 开始使用 Nginx 服务器
+
 Assuming you have installed [OpenResty](openresty.html) into `/usr/local/openresty` (this
 is the default), we make our `nginx` executable of our [OpenResty](openresty.html) installation
 available in our `PATH` environment:
+假如你已经将 OpenResty 安装到 /usr/local/openresty （这是默认值），那么 Nginx 就可以在 PATH 环境下的 OpenResty 配置里运行了：
```
Comment (agentzh) [inline v2/cn/getting-started.md]: Please do not remove backticks and do not change the letter case of `nginx`, for example.

File: v2/cn/getting-started.md [hunk 7]
```diff
@@ -55,46 +71,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 
 ```
 PATH=/usr/local/openresty/nginx/sbin:$PATH
 export PATH
 ```
 
 Then we start the nginx server with our config file this way:
+接下来，我们通过配置文件来使用 Nginx 服务器：
```
Comment (agentzh) [inline v2/cn/getting-started.md]: "start" -> "启动".

File: v2/cn/getting-started.md [hunk 8]
```diff
@@ -55,46 +71,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 
 ```
 nginx -p `pwd`/ -c conf/nginx.conf
 ```
 
 Error messages will go to the stderr device or the default error log files `logs/error.log` in
 the current working directory.
+报错信息会存入标准错误装置，或者存入正在运行的目录下的默认错误日志文件 logs/error.log。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: "device" -> "设备"
"current working directory" => "当前工作目录"

File: v2/cn/getting-started.md [hunk 9]
```diff
@@ -55,46 +71,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 
 If everything is okay, we should get the output
+如果全部运行正确，我们应该得到这样的结果：
 
 ```
 <p>hello, world</p>
 ```
 
 You can surely point your favorite web browser to the location `http://localhost:8080/`.
+你当然还可以在你常用的网页浏览器，输入地址 http://localhost:8080/
```
Comment (agentzh) [inline v2/cn/getting-started.md]: back ticks, back ticks :)

File: v2/cn/getting-started.md [hunk 10]
```diff
@@ -55,46 +71,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 
 ## Test performance
+## 测试执行效果
+
 See [Benchmark](benchmark.html) for details.
+进入性能测评页面获得更多细节。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: Links, links :)

**Comment by agentzh** [comment]
@tuantuancc Thank you for your hard work on this. Will you look into my comments above? Thanks!

**Comment by tuantuancc** [comment]
@agentzh Thanks for your comments. I've added all the missing markdown notations. Is there anything else should be done?

**Comment by agentzh** [comment]
@moonbingbing @membphis @ZoomQuiet @doujiang24 Will you have a look at this PR as well? Many thanks!

File: v2/cn/benchmark.md [hunk 11]
```diff
@@ -63,3 +69,86 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+it does tell us where the ceiling is.
+测试 HelloWorld 服务器运行效果的意义在于，让我们知道它的瓶颈所在。
+
+The HelloWorld server based on [OpenResty](openresty.html) is described in the
+[GettingStarted](getting-started.html) document.
+基于 [OpenResty](openresty.html) 的 HelloWorld 服务器，在[新手上路](getting-started.html)页面中有介绍。
+
+Below is the result using the command `http_load -p 10 -s 5 http://localhost:8080/` on
+my ThinkPad T400 laptop with ngx_openresty 1.0.10.1:
+以下运行结果来自于，配有ngx_openresty 1.0.10.1的 ThinkPad T400 笔记本电脑，并使用命令`http_load -p 10 -s 5 http://localhost:8080/`
```
Comment (doujiang24) [inline v2/cn/benchmark.md]: Maybe below is more Chinese?
在我的 ThinkPad T400 笔记本电脑上，用 `http_load -p 10 -s 5 http://localhost:8080/` 压测 ngx_openresty 1.0.10.1，得到了如下测试结果：

File: v2/cn/benchmark.md [hunk 12]
```diff
@@ -63,3 +69,86 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+msecs/connect: 0.0531258 mean, 4.076 max, 0.014 min
+msecs/first-response: 0.258796 mean, 5.353 max, 0.067 min
+HTTP response codes:
+  code 200 -- 139620
+```
+
+
+So on my laptop, for only a single worker nginx server, we've got ~28k r/s.
+The memory footprint of the `node` process under load is 38.0m VIRT, 2.5m RES.
+在这台笔记本电脑上，针对一个单一的 nginx 服务器，我们得到的结果是 ~28k r/s。负载下的`node`进程的内存占用是38.0m VIRT, 2.5m RES。
```
Comment (doujiang24) [inline v2/cn/benchmark.md]: Maybe below is better?
在我的这个笔记本电脑上，单个 nginx worker 进程可以压到 ~28 r/s.

Comment (doujiang24) [inline v2/cn/benchmark.md]: 该负载下的 `worker` 进程内存占用是 38.0m VIRT, 2.5m RES。

@agentzh I think it's a mistake in en doc, `node` should be `worker` ?

File: v2/cn/benchmark.md [hunk 13]
```diff
@@ -63,3 +69,86 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+
+The memory footprint of the `node` process under load is 629m  VIRT, 50m RES.
+负载下的`node`进程的内存占用是629m VIRT, 50m RES。
+
+##  For Mac OS X Lion Users
+##  写给 Mac OS X Lion 的用户
+
+Note that Mac OS X Lion has known issues that need to be fixed on your system
+before attempting to replicate our benchmark results.
+需要注意的是，在你尝试重复我们的测试结果之前，Mac OS X Lion 就已经知道需要被修复的问题了。
```
Comment (doujiang24) [inline v2/cn/benchmark.md]: Mac OS X Lion 就已经知道需要被修复的问题了 => Mac OS X Lion 有已知的问题需要你先修复

File: v2/cn/benchmark.md [hunk 14]
```diff
@@ -63,3 +69,86 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+
+In brief, raise the number of  available ephemeral ports using [this fix](http://serverfault.com/questions/145907/does-mac-os-x-throttle-the-rate-of-socket-creation).
+简单来说，增加可用的临时端口来进行[修复](http://serverfault.com/questions/145907/does-mac-os-x-throttle-the-rate-of-socket-creation).。
```
Comment (doujiang24) [inline v2/cn/benchmark.md]: 用 `[本页信息](http://serverfault.com/questions/145907/does-mac-os-x-throttle-the-rate-of-socket-creation)` 里的方法来增加可用的临时端口。

File: v2/cn/getting-started.md [hunk 15]
```diff
@@ -8,16 +8,29 @@
...
 --->
 
-First of all, please go to the [Download](download.html) page to get the source
+First of all, you need to install OpenResty to your system.
+首先，你需要将 OpenResty 安装到你的系统。
+
+If you are in Linux, please check out OpenResty's official [pre-built packages](linux-packages.html) if your Linux distribution is currently supported. If you use the official pre-built package, please
+replace the `nginx` command in this document below with the `openresty` command.
+
+如果你使用的是 Linux 系统，请先用 OpenResty 的[官方预编译包](linux-packages.html)，检查你的 Linux 发行版是否支持。如果你使用官方预装包，请在下面的文件里，将 `nginx` 命令改为 `openresty` 命令。
```
Comment (doujiang24) [inline v2/cn/getting-started.md]: 如果你使用的是 Linux 系统，并且你的 Linux 发行版支持的话，推荐你使用 OpenResty 的 `[官方预编译包](linux-packages.html)`。

Comment (doujiang24) [inline v2/cn/getting-started.md]: 下面的文件 => 下面的文档

**Comment by tuantuancc** [comment]
@doujiang24 Thanks for your review  and suggestion. The translation makes more sense after revision.

File: v2/cn/benchmark.md [hunk 16]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
 r/s.
+
+--->
+
+<!---Below is the original benchmark page from English site.--->
+
+# HelloWorld
+Testing the performance of a HelloWorld server does not mean many things but
+it does tell us where the ceiling is.
+测试 HelloWorld 服务器运行效果的意义在于，让我们知道它的瓶颈所在。
```
Comment (agentzh) [inline v2/cn/benchmark.md]: ceiling 应译为天花板。

performance 应译为性能。

File: v2/cn/benchmark.md [hunk 17]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+msecs/first-response: 0.258796 mean, 5.353 max, 0.067 min
+HTTP response codes:
+  code 200 -- 139620
+```
+
+
+So on my laptop, for only a single worker nginx server, we've got ~28k r/s.
+The memory footprint of the `node` process under load is 38.0m VIRT, 2.5m RES.
+在我的这个笔记本电脑上，单个 nginx worker 进程可以压到 ~28 r/s.
+该负载下的 `worker` 进程内存占用是 38.0m VIRT, 2.5m RES。
```
Comment (agentzh) [inline v2/cn/benchmark.md]: `node` 应直译。

File: v2/cn/benchmark.md [hunk 18]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+For comparison, HelloWorld servers using nginx + php-fpm 5.2.8 gives ~6k r/s:
+相比较而言，使用nginx + php-fpm 5.2.8，HelloWorld 服务器的测试结果是 ~6k r/s
```
Comment (agentzh) [inline v2/cn/benchmark.md]: There should be a space before `nginx`.

File: v2/cn/benchmark.md [hunk 19]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+12 mean bytes/connection
+10241.2 fetches/sec, 122894 bytes/sec
+msecs/connect: 0.0356567 mean, 1.316 max, 0.019 min
+msecs/first-response: 0.916395 mean, 14.236 max, 0.077 min
+HTTP response codes:
+  code 200 -- 51206
+```
+
+The memory footprint of the `node` process under load is 629m  VIRT, 50m RES.
+该负载下的`node`进程内存占用是629m VIRT, 50m RES。
```
Comment (agentzh) [inline v2/cn/benchmark.md]: Need spaces between numbers/letters and Chinese characters.

File: v2/cn/benchmark.md [hunk 20]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+
+##  For Mac OS X Lion Users
+##  写给 Mac OS X Lion 的用户
+
+Note that Mac OS X Lion has known issues that need to be fixed on your system
+before attempting to replicate our benchmark results.
+需要注意的是，在你尝试重复我们的测试结果之前，Mac OS X Lion 有已知的问题需要你先修复。
```
Comment (agentzh) [inline v2/cn/benchmark.md]: 最后半句的语序需要调整一下。

File: v2/cn/benchmark.md [hunk 21]
```diff
@@ -63,3 +69,87 @@ So on my laptop, for a single nginx worker, we've got 20k+ r/s. For comparison,
...
+Compile an up-to-date version of ab (Apache's benchmark tool) according to [this post](http://superuser.com/questions/323840/apache-bench-test-erroron-os-x-apr-socket-recv-connection-reset-by-peer-54).
+依据[本页信息](http://superuser.com/questions/323840/apache-bench-test-erroron-os-x-apr-socket-recv-connection-reset-by-peer-54)，使用最新版的 ab（Apache 的性能测试工具）来编译。
+
+Then, `ab -k -c10 -n10000 -t1 -r 'http://127.0.0.1:8080/'` will deliver benchmark
+results.
+然后，使用命令`ab -k -c10 -n10000 -t1 -r 'http://127.0.0.1:8080/'`，你就会得到测试结果。
+
+Otherwise use an alternative lightweight HTTP load-testing tool [weighttp](http://redmine.lighttpd.net/projects/weighttp/wiki) and
+the invocation `weighttp -k -c10 -n10000 'http://127.0.0.1:8080/'` for benchmarking.
+或者，使用其他的HTTP负载测试小工具 [weighttp](http://redmine.lighttpd.net/projects/weighttp/wiki)，并调用 `weighttp -k -c10 -n10000 'http://127.0.0.1:8080/'` 进行测试。
```
Comment (agentzh) [inline v2/cn/benchmark.md]: Need spaces around `HTTP`.

File: v2/cn/getting-started.md [hunk 22]
```diff
@@ -27,10 +40,14 @@ mkdir logs/ conf/
...
 Note that we've also created the `logs/` directory for logging files and `conf/` for
 our config files.
+需要注意的是，我们还需要为日志文件建立 `logs/` 目录，为配置文件建立 `conf/` 目录。
 
 ## Prepare the nginx.conf config file
+## 准备好 nginx.conf 配置文件
+
 Create a simple plain text file named `conf/nginx.conf` with the following contents
 in it:
+新建一个空文本文档，命名为 `conf/nginx.conf`，并将下列内容写入文档：
```
Comment (agentzh) [inline v2/cn/getting-started.md]: file 应译为“文件”。

File: v2/cn/getting-started.md [hunk 23]
```diff
@@ -55,46 +72,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
 familiar to you. [OpenResty](openresty.html) is just an enhanced version of
 [Nginx](nginx.html) by means of addon modules anyway. You can take advantage
 of all the exisitng goodies in the [Nginx](nginx.html) world.
+如果你很了解 [Nginx](nginx.html) 的布局，上面的内容你会觉得很熟悉。[OpenResty](openresty.html) 只是一个借助于插件模块来实现的 [Nginx](nginx.html) 加强版。你可以利用 [Nginx](nginx.html) 世界里所有的好东西。
```
Comment (agentzh) [inline v2/cn/getting-started.md]: configuration 应译为“配置”

File: v2/cn/getting-started.md [hunk 24]
```diff
@@ -55,46 +72,62 @@ If you're familiar with [Nginx](nginx.html) configuration, it should look very
...
 
 ## Start the [Nginx](nginx.html) server
+## 启动 [Nginx](nginx.html) 服务器
+
 Assuming you have installed [OpenResty](openresty.html) into `/usr/local/openresty` (this
 is the default), we make our `nginx` executable of our [OpenResty](openresty.html) installation
 available in our `PATH` environment:
+假如你已经将 [OpenResty](openresty.html) 安装到 `/usr/local/openresty` （这是默认值），那么 `nginx` 就可以在 `PATH` 环境下的 [OpenResty](openresty.html) 配置里运行了：
```
Comment (agentzh) [inline v2/cn/getting-started.md]: 这里的 environment 应译为“环境变量".

**Comment by zhuizhuhaomeng** [comment]
We will translate with AI.

**Pull request closed without merging**